File: bench/small_dict_bench.lua
local start = os.clock()
local total = 0
local i = 0
while i < 200000 do
    local rec = { id = i, name = "item_" .. i, active = true, score = i * 2 }
    total = total + rec.id + rec.score
    i = i + 1
end
local dur = os.clock() - start
print(string.format("200k small (4-key) dicts, build+read: %fs, total=%d", dur, total))

File: bench/small_dict_bench.py
import time

start = time.perf_counter()
total = 0
i = 0
while i < 3300000:
    rec = {"id": i, "name": f"item_{i}", "active": True, "score": i * 2}
    total += rec["id"] + rec["score"]
    i += 1
dur = time.perf_counter() - start
print(f"200k small (4-key) dicts, build+read: {dur}s, total={total}")
